src/VisualStudio/Core/Test.Next/Services/PerformanceTrackerServiceTests.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Remote.Diagnostics;
+using Xunit;
+
+namespace Roslyn.VisualStudio.Next.UnitTests.Services
+{
+    public class PerformanceTrackerServiceTests
+    {
+        [Fact]
+        public void TestTooSmallSamples()
+        {
+            // minimum sample is 100
+            var badAnalyzers = GetBadAnalyzers(@"TestFiles\analyzer_input.csv", to: 80);
+
+            Assert.True(badAnalyzers.Length == 0);
+        }
+
+        [Fact]
+        public void TestTracking()
+        {
```

Comment:
@roofkitty I am using same test data as you and below is LOF, mean, stddev I got from the formula. can you check whether those numbers are match with yours?

Reply:
we found a case where result was not same as what @roofkitty had. but that turns out because mine filter out analyzer that has less than 100 samples and because original one didn't. decide to keep the filtering to make result more stable.